[BUG] Stack overflow with badly formed PostConstruct method

## Describe the Bug

While porting a Spring application to Avaje Inject, we encountered a crash in the annotation processor.

## Environment Info
12.5-RC2
25.0.2
Maven

## Minimal Reproducible Example

```java
import jakarta.annotation.PostConstruct;
import jakarta.inject.Singleton;

@Singleton
public class BadSingleton {
    @PostConstruct
    public BadSingleton badPostConstruct() {
        return null;
    }
}
```

## Expected Behavior

Avaje should either refuse to process this with a clear error message, or ignore the return type as unimportant to the container.

## Actual Behavior

Compiler crash:

```
Caused by: java.lang.StackOverflowError
    at com.sun.tools.javac.code.Symbol.apiComplete (Symbol.java:709)
    at com.sun.tools.javac.code.Symbol.getAnnotationMirrors (Symbol.java:753)
    at com.sun.tools.javac.code.Symbol$TypeSymbol.getAnnotationMirrors (Symbol.java:818)
    at io.avaje.inject.generator.AspectPrism.getMirror (AspectPrism.java:207)
    at io.avaje.inject.generator.AspectPrism.getInstanceOn (AspectPrism.java:65)
    at io.avaje.inject.generator.TypeExtendsInjection.readAspects (TypeExtendsInjection.java:69)
    at io.avaje.inject.generator.TypeExtendsInjection.<init> (TypeExtendsInjection.java:43)
    at io.avaje.inject.generator.TypeExtendsReader.<init> (TypeExtendsReader.java:55)
    at io.avaje.inject.generator.TypeReader.<init> (TypeReader.java:57)
    at io.avaje.inject.generator.TypeReader.<init> (TypeReader.java:41)
    at io.avaje.inject.generator.MethodReader.<init> (MethodReader.java:149)
    at io.avaje.inject.generator.MethodReader.<init> (MethodReader.java:68)
    at io.avaje.inject.generator.MethodLifecycleReader.<init> (MethodLifecycleReader.java:24)
    at io.avaje.inject.generator.MethodReader.<init> (MethodReader.java:151)
    at io.avaje.inject.generator.MethodReader.<init> (MethodReader.java:68)
    at io.avaje.inject.generator.MethodLifecycleReader.<init> (MethodLifecycleReader.java:24)
    at io.avaje.inject.generator.MethodReader.<init> (MethodReader.java:151)
    at io.avaje.inject.generator.MethodReader.<init> (MethodReader.java:68)
```

--- a/blackbox-test-inject/src/main/java/org/example/myapp/lifecycle/PostConstructReturnsThis.java
+++ b/blackbox-test-inject/src/main/java/org/example/myapp/lifecycle/PostConstructReturnsThis.java
@@ -1,4 +1,4 @@
-package io.avaje.inject.generator.models.valid.lifecycle;
+package org.example.myapp.lifecycle;
 
 import io.avaje.inject.PostConstruct;
 import jakarta.inject.Singleton;

--- a/blackbox-test-inject/src/test/java/org/example/myapp/lifecycle/PostConstructReturnsThisTest.java
+++ b/blackbox-test-inject/src/test/java/org/example/myapp/lifecycle/PostConstructReturnsThisTest.java
@@ -1,0 +1,20 @@
+package org.example.myapp.lifecycle;
+
+import io.avaje.inject.BeanScope;
+import org.junit.jupiter.api.Test;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+class PostConstructReturnsThisTest {
+
+  @Test
+  void beanWithPostConstructReturningSelf() {
+    try (BeanScope beanScope = BeanScope.builder().build()) {
+      var bean = beanScope.get(PostConstructReturnsThis.class);
+      assertThat(bean).isNotNull();
+
+      var self = bean.init();
+      assertThat(bean).isSameAs(self);
+    }
+  }
+}

--- a/inject-generator/src/main/java/io/avaje/inject/generator/MethodReader.java
+++ b/inject-generator/src/main/java/io/avaje/inject/generator/MethodReader.java
@@ -149,13 +149,14 @@ final class MethodReader {
       this.typeReader = new TypeReader(beanTypes.orElse(List.of()), genericType, returnElement, importTypes, element);
       typeReader.process();
       if (bean != null) {
+        // factory @Bean method, read lifecycle methods from the return type
         var lifecycleReader = new MethodLifecycleReader(returnElement, initMethod, destroyMethod, importTypes);
         this.initMethod = lifecycleReader.initMethod();
         this.initMethodReader = lifecycleReader.initMethodReader();
         this.destroyMethod = lifecycleReader.destroyMethod();
       } else {
-        this.initMethod = initMethod;
-        this.destroyMethod = destroyMethod;
+        this.initMethod = null;
+        this.destroyMethod = null;
       }
     }
     if (lazy && prototype) {

